### PR TITLE
fix(rules): canonical paths in aliases

### PR DESCRIPTION
**Pull request by rgrinberg**
The canonical paths were all wrong for modules with (include_subdirs
qualified). This PR addresses the problem.

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

File: src/dune_rules/modules.ml [hunk 1]
```diff
@@ -1131,3 +1166,19 @@ let source_dirs =
       Module.sources m
       |> List.fold_left ~init:acc ~f:(fun acc f ->
              Path.Set.add acc (Path.parent_exn f)))
+
+let canonical_path t (group : Group.t) m =
+  (* Format.eprintf "module: %s@.%!" (Dyn.to_string (Module.to_dyn m)); *)
```
Comment (Alizter) [inline src/dune_rules/modules.ml:1171]: leftovers

File: src/dune_rules/modules.ml [hunk 2]
```diff
@@ -477,34 +483,42 @@ module Group = struct
       match Module_name.Map.find modules name with
       | Some (Group g) -> g
       | Some (Module m) ->
-        Code_error.raise "for_alias: unexpected module"
+        Code_error.raise "group_of_alias: unexpected module"
```
Comment (anmonteiro) [inline src/dune_rules/modules.ml:486]: isn't this "group for alias" instead?

Comment (anmonteiro) [reply src/dune_rules/modules.ml:486]: ah I get it. It's finding the group for the alias

File: src/dune_rules/modules.ml [hunk 3]
```diff
@@ -1131,3 +1166,14 @@ let source_dirs =
       Module.sources m
       |> List.fold_left ~init:acc ~f:(fun acc f ->
              Path.Set.add acc (Path.parent_exn f)))
+
+let canonical_path t (group : Group.t) m =
+  let path =
+    let path = Module.path m in
+    match Module_name.Map.find group.modules (Module.name m) with
+    | None | Some (Group.Module _) -> path
+    | Some (Group _) -> path |> List.rev |> List.tl |> List.rev
```
Comment (anmonteiro) [inline src/dune_rules/modules.ml:1175]: what's this case here? why are we removing the last path segment?

**Review by anmonteiro — APPROVED** [review]
I have a question about the implementation, but tests are passing, so approving

File: src/dune_rules/modules.ml (hunk 3 above)
Comment (rgrinberg) [reply src/dune_rules/modules.ml:1175]: It's for interface modules. A module like `foo/foo.ml` would have a path like `[ "Foo"; "Foo" ]` so the last element has to be removed to get the canonical path.

**Pull request merged**